src/extensions/default/SVGCodeHints/SVGUtils.js

```diff
@@ -0,0 +1,339 @@
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, brackets, $ */
+
+define(function (require, exports, module) {
+    "use strict";
+    
+    // Load dependencies.
+    var TokenUtils  = brackets.getModule("utils/TokenUtils");
+    
+    // Enums of token types.
+    var TokenTag    = 1,
+        TokenAttr   = 2,
+        TokenValue  = 3;
+    
+    // Regex to find whether or not an attribute value is quoted in single or double quotation marks.
+    var regexQuotedValue = /^['"][\W\w]*?['"]?$/;
+    
+    // Regex to find whitespace.
+    var regexWhitespace = /^\s+?$/;
+    
+    /**
+     * Returns an object that represents all its params.
+     *
+     * @param {Token} token
+     * Token at the current pos
+     * 
+     * @param {int} tokenType
+     * Type of current token
+     *
+     * @param {int} offset
+     * Offset in current token
+     *
+     * @param {Array.<string>} exclusionList
+     * List of attributes of a tag or attribute options used by an attribute
+     *
+     * @param {string} tagName
+     * Name of the current tag
+     *
+     * @param {string} attrName
+     * Name of the curent attribute
+     *
+     * @param {boolean} shouldReplace
+     * true if we don't want to append ="" to an attribute
+     *
+     * @return {token: Token, tokenType: int, offset: int, exclusionList: Array.<string>, tagName: string, attrName: string, shouldReplace: boolean}
+     */
+    function _createTagInfo(token, tokenType, offset, exclusionList, tagName, attrName, shouldReplace) {
+        return {
+            token: token || null,
+            tokenType: tokenType || null,
+            offset: offset || 0,
+            exclusionList: exclusionList || [],
+            tagName: tagName || "",
+            attrName: attrName || "",
+            shouldReplace: shouldReplace || false
+        };
+    }
+    
+    /**
+     * Return the tagName and a list of attributes used by the tag.
+     *
+     * @param {Editor} editor
+     * An instance of active editor
+     *
+     * @param {line: int, ch: int} constPos
+     * The position of cursor in the active editor
+     *
+     * @return {tagName: string, exclusionList: Array.<string>, shouldReplace: boolean}
+     */
+    function _getTagAttributes(editor, constPos) {
+        var pos, ctx, ctxPrev, ctxNext, ctxTemp, tagName, exclusionList = [], shouldReplace;
+        
+        pos = $.extend({}, constPos);
```

Comment:
Note that since `constPos` is multiple levels deep, I think you need to do a deep copy:

```js
        pos = $.extend(true, {}, constPos);
```